File: algorithms/nowcoder/huawei/eight.js
function solution() {
  let set = new Set();
  let map = new Map();
  let list = [];

  let n = +(readline());
  while (n-- > 0) {
    let temp = readline().split(" ").map(val => +val);
    if (map.has(temp[0])) {
      map.set(temp[0], map.get(temp[0]) + temp[1]);
    } else {
      map.set(temp[0], temp[1]);
    }
    if (!set.has(temp[0])) {
      set.add(temp[0]);
      list.push(temp[0]);
    }
  }
  list.sort((a, b) => a - b);
  for (let num of list) {
    console.log(num + " " + map.get(num));
  }
}

solution();

File: src/nowcoder/huawei/Eight.java
package src.nowcoder.huawei;

import java.util.*;
import java.io.*;

/**
 * @author Heming
 * created by IntelliJ IDEA
 * founded at 2021/04/09 20:50:37
 * description:
 */
public class Eight {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        List<Integer> list = new ArrayList<>();
        Set<Integer> record = new HashSet<>();
        Map<Integer, Integer> map = new HashMap<>();
        while (n-- > 0) {
            String[] in = br.readLine().split(" ");
            int first = Integer.parseInt(in[0]), second = Integer.parseInt(in[1]);
            if (map.containsKey(first)) {
                map.put(first, map.get(first) + second);
            } else {
                map.put(first, second);
            }
            if (record.add(first)) {
                list.add(first);
            }
        }
        Collections.sort(list);
        for (int i : list) {
            System.out.println(i + " " + map.get(i));
        }
    }
}
